frontends/concrete-python/examples/levenshtein_distance/levenshtein_distance.py

```diff
@@ -0,0 +1,366 @@
+    list_patterns = []
+    for _ in range(nb_strings):
+        for length_1 in range(len_min, len_max + 1):
+            for length_2 in range(len_min, len_max + 1):
+                list_patterns += [
+                    (
+                        random_string(mapping_to_int, length_1),
+                        random_string(mapping_to_int, length_2),
+                    )
+                    for _ in range(1)
+                ]
+
+    return list_patterns
+
+
+def compute_in_simulation(my_module, list_patterns, mapping_to_int):
+    """Check equality between distance in simulation and clear distance."""
+    print("Computations in simulation\n")
+
+    for a, b in list_patterns:
+
+        print(f"    Computing Levenshtein between strings '{a}' and '{b}'", end="")
+
+        a_as_int = tuple([mapping_to_int[ai] for ai in a])
+        b_as_int = tuple([mapping_to_int[bi] for bi in b])
+
+        l1_simulate = levenshtein_simulate(my_module, a_as_int, b_as_int)
+        l1_clear = levenshtein_clear(a_as_int, b_as_int)
+
+        assert l1_simulate == l1_clear, f"    {l1_simulate=} and {l1_clear=} are different"
+        print(" - OK")
+
+
+def compute_in_fhe(my_module, list_patterns, mapping_to_int, verbose=True, show_distance=False):
+    """Check equality between distance in FHE and clear distance."""
+    my_module.keygen()
+
+    # Checks in FHE
+    if verbose:
+        print("\nComputations in FHE\n")
+
+    for a, b in list_patterns:
+
+        print(f"    Computing Levenshtein between strings '{a}' and '{b}'", end="")
+
+        a_as_int = [mapping_to_int[ai] for ai in a]
+        b_as_int = [mapping_to_int[bi] for bi in b]
+
+        a_enc = tuple(my_module.equal.encrypt(ai, None)[0] for ai in a_as_int)
+        b_enc = tuple(my_module.equal.encrypt(None, bi)[1] for bi in b_as_int)
+
+        time_begin = time.time()
+        l1_fhe_enc = levenshtein_fhe(my_module, a_enc, b_enc)
+        time_end = time.time()
+
+        l1_fhe = my_module.mix.decrypt(l1_fhe_enc)
+
+        l1_clear = levenshtein_clear(a, b)
+
+        assert l1_fhe == l1_clear, f"    {l1_fhe=} and {l1_clear=} are different"
+
+        if not show_distance:
+            print(f" - OK in {time_end - time_begin:.2f} seconds")
+        else:
+            print(f" - distance is {l1_fhe}, computed in {time_end - time_begin:.2f} seconds")
+
+
+def main():
+    """Main function."""
+    print()
+
+    # Options by the user
+    args = manage_args()
+
+    # Do what the user requested
+    if args.autotest:
+        mapping_to_int = prepare_alphabet_mapping(args.alphabet)
+        my_module = compile_module(mapping_to_int, args)
+        list_patterns = prepare_random_patterns(mapping_to_int, 0, args.max_string_length, 1)
+        compute_in_simulation(my_module, list_patterns, mapping_to_int)
+        compute_in_fhe(my_module, list_patterns, mapping_to_int)
+        print("")
+
+    if args.autoperf:
+        for alphabet in ["ACTG", "string", "STRING", "StRiNg"]:
```

Comment:
I'd really prefer to have a class for the alphabet with `encode`, `decode`, `check`, ... methods. It'd make the example much clear IMO.

Also you would be able to do:
```python
for alphabet in [Alphabet.lowercase(), Alphabet.uppercase(), Alphabet.dna(), ...]:
```
Which is less error prone and more clear IMO.

Reply:
I'm not a big fan of OOP, especially when like here, it doesn't seem needed. But if you want really it, I'll do. (And for sure, I have not the right habits for OOP so I am going to make something not idomatic, you'll correct me)

Reply:
I don't think this is OOP. Sure it has a bit of encapsulation and maybe some abstraction, but there is no inheritance or polymorphism. It's just a way to make alphabets more type safe and understandable.

Would be better to me, wdyt @BourgerieQuentin?

Reply:
I've tried it, tell me what you think